modules/objdetect/src/qrcode.cpp

```diff
@@ -2359,10 +2418,86 @@ bool QRDecode::versionDefinition()
         }
     }
     version = saturate_cast<uint8_t>((std::min(transition_x, transition_y) - 1) * 0.25 - 1);
+    if (version_tmp >= 7 || version >= 7) {
```

Comment:
I propose to rename version candidates to proper names like `versionByNumModules`, `versionByNumTransitions` or something else meaningful.